### PR TITLE
feat: add `WindowBuilder::with_focused`

### PR DESCRIPTION
**Pull request by amrbashir**
ref: https://github.com/tauri-apps/tauri/issues/5120

### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Review by keiya01 — APPROVED** [review]
LGTM on macos

**Comment by lucasfernog** [comment]
@amrbashir this doesn't really work on Linux :( `accepts_focus` forces the window to **never** receive focus :(

**Comment by amrbashir** [comment]
pushed a little workaround, hope that fixes it.

**Comment by amrbashir** [comment]
Doesn't work on Linux at all

File: src/platform_impl/linux/window.rs [hunk 1]
```diff
@@ -242,10 +242,6 @@ impl Window {
       }
     }
 
-    if let Parent::ChildOf(parent) = pl_attribs.parent {
-      window.set_transient_for(Some(&parent));
-    }
-
```
Comment (amrbashir) [inline src/platform_impl/linux/window.rs]: why remove this?

**Review by lucasfernog — APPROVED** [review]

File: src/platform_impl/linux/window.rs (hunk 1 above)
Comment (lucasfernog) [reply src/platform_impl/linux/window.rs]: i thought it was part of your workaround, it's not needed for mine at least

Comment (amrbashir) [reply src/platform_impl/linux/window.rs]: no that's a different feature. I just reordered it.

File: src/window.rs [hunk 2]
```diff
@@ -411,6 +415,13 @@ impl WindowBuilder {
     self
   }
 
+  /// Whether the window will be initially focused or not.
```
Comment (amrbashir) [inline src/window.rs:422]:
```suggestion
  /// Whether the window will be initially focused or not.
  ///
  /// ## Platform-specific:
  ///
  /// **Android / iOS:** Unsupported.
```

File: src/window.rs [hunk 3]
```diff
@@ -208,6 +208,9 @@ pub struct WindowAttributes {
   pub window_menu: Option<platform_impl::Menu>,
 
   pub preferred_theme: Option<Theme>,
+
+  /// Whether the window should be initially focused or not.
```
Comment (amrbashir) [inline src/window.rs:212]:
```suggestion
  /// Whether the window should be initially focused or not.
  ///
  /// ## Platform-specific:
  ///
  /// **Android / iOS:** Unsupported.
```

File: src/platform_impl/linux/window.rs [hunk 4]
```diff
@@ -242,8 +248,19 @@ impl Window {
       window.hide();
     }
 
-    if let Parent::ChildOf(parent) = pl_attribs.parent {
-      window.set_transient_for(Some(&parent));
+    // restore accept-focus after the window has been drawn
+    // if the window was initially created without focus
```
Comment (amrbashir) [inline src/platform_impl/linux/window.rs:256]:
```suggestion
	if let Parent::ChildOf(parent) = pl_attribs.parent {
      window.set_transient_for(Some(&parent));
    }
    
    // restore accept-focus after the window has been drawn
    // if the window was initially created without focus
```

File: src/platform_impl/linux/window.rs (hunk 1 above)
Comment (lucasfernog) [reply src/platform_impl/linux/window.rs]: so why is it in this PR?

Comment (amrbashir) [reply src/platform_impl/linux/window.rs]: I just moved it up by mistake while working on the workaround. Should be good now.

Comment (lucasfernog) [reply src/platform_impl/linux/window.rs]: ahh i understood now, alright

**Pull request merged**